core/src/main/java/bio/terra/pearl/core/service/search/EnrolleeSearchExpressionParser.java

```diff
@@ -0,0 +1,171 @@
+package bio.terra.pearl.core.service.search;
+
+import bio.terra.pearl.core.antlr.CohortRuleLexer;
+import bio.terra.pearl.core.antlr.CohortRuleParser;
+import bio.terra.pearl.core.dao.participant.EnrolleeDao;
+import bio.terra.pearl.core.dao.participant.MailingAddressDao;
+import bio.terra.pearl.core.dao.participant.ProfileDao;
+import bio.terra.pearl.core.dao.survey.AnswerDao;
+import bio.terra.pearl.core.service.rule.RuleParsingErrorListener;
+import bio.terra.pearl.core.service.search.expressions.BooleanSearchExpression;
+import bio.terra.pearl.core.service.search.expressions.ComparisonOperator;
+import bio.terra.pearl.core.service.search.expressions.DefaultSearchExpression;
+import bio.terra.pearl.core.service.search.expressions.EnrolleeTermComparisonFacet;
+import bio.terra.pearl.core.service.search.terms.AgeTerm;
+import bio.terra.pearl.core.service.search.terms.AnswerTerm;
+import bio.terra.pearl.core.service.search.terms.EnrolleeTerm;
+import bio.terra.pearl.core.service.search.terms.ProfileTerm;
+import bio.terra.pearl.core.service.search.terms.SearchValue;
+import bio.terra.pearl.core.service.search.terms.UserInputTerm;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.misc.ParseCancellationException;
+import org.apache.commons.lang3.StringUtils;
+import org.jooq.Operator;
+import org.springframework.stereotype.Component;
+
+@Component
+public class EnrolleeSearchExpressionParser {
+    private final EnrolleeDao enrolleeDao;
+    private final AnswerDao answerDao;
+    private final ProfileDao profileDao;
+    private final MailingAddressDao mailingAddressDao;
+
+
+    public EnrolleeSearchExpressionParser(EnrolleeDao enrolleeDao, AnswerDao answerDao, ProfileDao profileDao, MailingAddressDao mailingAddressDao) {
+        this.enrolleeDao = enrolleeDao;
+        this.answerDao = answerDao;
+        this.profileDao = profileDao;
+        this.mailingAddressDao = mailingAddressDao;
+    }
+
+
+    public EnrolleeSearchExpression parseRule(String rule) {
+        if (StringUtils.isBlank(rule)) {
+            return new DefaultSearchExpression(enrolleeDao, profileDao);
+        }
+
+        try {
+            CohortRuleLexer lexer = new CohortRuleLexer(CharStreams.fromString(rule));
+            lexer.removeErrorListeners();
+            lexer.addErrorListener(new RuleParsingErrorListener());
+            CommonTokenStream tokens = new CommonTokenStream(lexer);
+            CohortRuleParser parser = new CohortRuleParser(tokens);
+            parser.removeErrorListeners();
+            parser.addErrorListener(new RuleParsingErrorListener());
+            CohortRuleParser.ExprContext exp = parser.expr();
+
+            return parseExpression(exp);
+        } catch (ParseCancellationException e) {
+            throw new IllegalArgumentException("Error parsing rule: " + e.getMessage());
```

Comment:
```suggestion
            throw new RuleParsingException("Error parsing rule: " + e.getMessage());
```

Reply:
I think rule parsing errors warrant their own exception class--this will let the backend communicate to the frontend exactly why, e.g., the survey failed to save.  We have RuleParsingException for this -- if you want that to be an unchecked exception, you can update that class to extend IllegalArgumentException